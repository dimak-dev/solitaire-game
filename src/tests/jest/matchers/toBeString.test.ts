import MatcherContext = jest.MatcherContext;
import toBeString from "Tests/jest/matchers/toBeString";

describe('toBeString Custom Matcher', () => {
    let matcherContextMock: MatcherContext;
    beforeEach(() => {
        matcherContextMock = {
            // @ts-ignore
            utils: {
                printReceived: jest.fn().mockImplementation((...args) => `printReceived mock, args: ${JSON.stringify(args)}`),
                printExpected: jest.fn().mockImplementation((...args) => `printExpected mock, args: ${JSON.stringify(args)}`),
                matcherHint: jest.fn().mockImplementation((...args) => `matcherHint mock, args: ${JSON.stringify(args)}`)
            }
        };
    });

    test('Empty primitive string is a string', () => {
        const result = toBeString.call(matcherContextMock, '');
        expect(result.pass).toBeTruthy();
        expect(result.message()).toMatchSnapshot();
    });

    test('Empty string as String object is a string', () => {
        const result = toBeString.call(matcherContextMock, String(''));
        expect(result.pass).toBeTruthy();
        expect(result.message()).toMatchSnapshot();
    });

    test('Not empty primitive string with letters is a string', () => {
        const result = toBeString.call(matcherContextMock, 'letters');
        expect(result.pass).toBeTruthy();
        expect(result.message()).toMatchSnapshot();
    });

    test('Not empty string as String object with letters is a string', () => {
        const result = toBeString.call(matcherContextMock, String('letters'));
        expect(result.pass).toBeTruthy();
        expect(result.message()).toMatchSnapshot();
    });

    test('Not empty primitive string with zero is a string', () => {
        const result = toBeString.call(matcherContextMock, '0');
        expect(result.pass).toBeTruthy();
        expect(result.message()).toMatchSnapshot();
    });

    test('Not empty string as String object with zero is a string', () => {
        const result = toBeString.call(matcherContextMock, String('0'));
        expect(result.pass).toBeTruthy();
        expect(result.message()).toMatchSnapshot();
    });

    test('Not empty primitive string with numbers is a string', () => {
        const result = toBeString.call(matcherContextMock, '123456');
        expect(result.pass).toBeTruthy();
        expect(result.message()).toMatchSnapshot();
    });

    test('Not empty string as String object with numbers is a string', () => {
        const result = toBeString.call(matcherContextMock, String('123456'));
        expect(result.pass).toBeTruthy();
        expect(result.message()).toMatchSnapshot();
    });

    test('number as primitive is not a string', () => {
        const result = toBeString.call(matcherContextMock, 123456);
        expect(result.pass).toBeFalsy();
        expect(result.message()).toMatchSnapshot();
    });

    test('number as Number object is not a string', () => {
        const result = toBeString.call(matcherContextMock, Number(123456));
        expect(result.pass).toBeFalsy();
        expect(result.message()).toMatchSnapshot();
    });

    test('zero number as primitive is not a string', () => {
        const result = toBeString.call(matcherContextMock, 0);
        expect(result.pass).toBeFalsy();
        expect(result.message()).toMatchSnapshot();
    });

    test('zero number as Number object is not a string', () => {
        const result = toBeString.call(matcherContextMock, Number(0));
        expect(result.pass).toBeFalsy();
        expect(result.message()).toMatchSnapshot();
    });

    test('number as Bigint object is not a string', () => {
        const result = toBeString.call(matcherContextMock, BigInt(123456));
        expect(result.pass).toBeFalsy();
        expect(result.message()).toMatchSnapshot();
    });

    test('extra large number as Bigint object is not a string', () => {
        const result = toBeString.call(matcherContextMock, BigInt(1234567890123456789012345));
        expect(result.pass).toBeFalsy();
        expect(result.message()).toMatchSnapshot();
    });

    test('zero number as Bigint object is not a string', () => {
        const result = toBeString.call(matcherContextMock, BigInt(0));
        expect(result.pass).toBeFalsy();
        expect(result.message()).toMatchSnapshot();
    });

    test('true boolean as primitive is not a string', () => {
        const result = toBeString.call(matcherContextMock, true);
        expect(result.pass).toBeFalsy();
        expect(result.message()).toMatchSnapshot();
    });

    test('false boolean as primitive is not a string', () => {
        const result = toBeString.call(matcherContextMock, false);
        expect(result.pass).toBeFalsy();
        expect(result.message()).toMatchSnapshot();
    });

    test('true boolean as object is not a string', () => {
        const result = toBeString.call(matcherContextMock, Boolean(true));
        expect(result.pass).toBeFalsy();
        expect(result.message()).toMatchSnapshot();
    });

    test('false boolean as object is not a string', () => {
        const result = toBeString.call(matcherContextMock, Boolean(false));
        expect(result.pass).toBeFalsy();
        expect(result.message()).toMatchSnapshot();
    });

    test('undefined is not a string', () => {
        const result = toBeString.call(matcherContextMock, undefined);
        expect(result.pass).toBeFalsy();
        expect(result.message()).toMatchSnapshot();
    });

    test('null is not a string', () => {
        const result = toBeString.call(matcherContextMock, null);
        expect(result.pass).toBeFalsy();
        expect(result.message()).toMatchSnapshot();
    });

    test('Symbol is not a string', () => {
        const result = toBeString.call(matcherContextMock, Symbol());
        expect(result.pass).toBeFalsy();
        expect(result.message()).toMatchSnapshot();
    });

    test('Symbol with description is not a string', () => {
        const result = toBeString.call(matcherContextMock, Symbol('description'));
        expect(result.pass).toBeFalsy();
        expect(result.message()).toMatchSnapshot();
    });

    test('Empty object is not a string', () => {
        const result = toBeString.call(matcherContextMock, {});
        expect(result.pass).toBeFalsy();
        expect(result.message()).toMatchSnapshot();
    });

    test('object with string value is not a string', () => {
        const result = toBeString.call(matcherContextMock, {key: 'value'});
        expect(result.pass).toBeFalsy();
        expect(result.message()).toMatchSnapshot();
    });

    test('empty array is not a string', () => {
        const result = toBeString.call(matcherContextMock, []);
        expect(result.pass).toBeFalsy();
        expect(result.message()).toMatchSnapshot();
    });

    test('array with string value is not a string', () => {
        const result = toBeString.call(matcherContextMock, ['value']);
        expect(result.pass).toBeFalsy();
        expect(result.message()).toMatchSnapshot();
    });
});