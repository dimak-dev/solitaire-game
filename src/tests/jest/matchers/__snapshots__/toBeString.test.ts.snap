// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toBeString Custom Matcher Empty object is not a string 1`] = `
"matcherHint mock, args: [".toBeString","received",""]

Expected:
  printExpected mock, args: ["type of string OR instanceof String"]
Received:
  printReceived mock, args: ["object"]"
`;

exports[`toBeString Custom Matcher Empty primitive string is a string 1`] = `
"matcherHint mock, args: [".not.toBeString","received",""]

Expected value must be not type of string, but string is received:
  printReceived mock, args: [""]"
`;

exports[`toBeString Custom Matcher Empty string as String object is a string 1`] = `
"matcherHint mock, args: [".not.toBeString","received",""]

Expected value must be not type of string, but string is received:
  printReceived mock, args: [""]"
`;

exports[`toBeString Custom Matcher Not empty primitive string with letters is a string 1`] = `
"matcherHint mock, args: [".not.toBeString","received",""]

Expected value must be not type of string, but string is received:
  printReceived mock, args: ["letters"]"
`;

exports[`toBeString Custom Matcher Not empty primitive string with numbers is a string 1`] = `
"matcherHint mock, args: [".not.toBeString","received",""]

Expected value must be not type of string, but string is received:
  printReceived mock, args: ["123456"]"
`;

exports[`toBeString Custom Matcher Not empty primitive string with zero is a string 1`] = `
"matcherHint mock, args: [".not.toBeString","received",""]

Expected value must be not type of string, but string is received:
  printReceived mock, args: ["0"]"
`;

exports[`toBeString Custom Matcher Not empty string as String object with letters is a string 1`] = `
"matcherHint mock, args: [".not.toBeString","received",""]

Expected value must be not type of string, but string is received:
  printReceived mock, args: ["letters"]"
`;

exports[`toBeString Custom Matcher Not empty string as String object with numbers is a string 1`] = `
"matcherHint mock, args: [".not.toBeString","received",""]

Expected value must be not type of string, but string is received:
  printReceived mock, args: ["123456"]"
`;

exports[`toBeString Custom Matcher Not empty string as String object with zero is a string 1`] = `
"matcherHint mock, args: [".not.toBeString","received",""]

Expected value must be not type of string, but string is received:
  printReceived mock, args: ["0"]"
`;

exports[`toBeString Custom Matcher Not empty string by String() constructor with letters is a string 1`] = `
"matcherHint mock, args: [".not.toBeString","received",""]

Expected value must be not type of string, but string is received:
  printReceived mock, args: ["letters"]"
`;

exports[`toBeString Custom Matcher Not empty string by String() constructor with numbers is a string 1`] = `
"matcherHint mock, args: [".not.toBeString","received",""]

Expected value must be not type of string, but string is received:
  printReceived mock, args: ["123456"]"
`;

exports[`toBeString Custom Matcher Not empty string by String() constructor with zero is a string 1`] = `
"matcherHint mock, args: [".not.toBeString","received",""]

Expected value must be not type of string, but string is received:
  printReceived mock, args: ["0"]"
`;

exports[`toBeString Custom Matcher Symbol is not a string 1`] = `
"matcherHint mock, args: [".toBeString","received",""]

Expected:
  printExpected mock, args: ["type of string OR instanceof String"]
Received:
  printReceived mock, args: ["symbol"]"
`;

exports[`toBeString Custom Matcher Symbol with description is not a string 1`] = `
"matcherHint mock, args: [".toBeString","received",""]

Expected:
  printExpected mock, args: ["type of string OR instanceof String"]
Received:
  printReceived mock, args: ["symbol"]"
`;

exports[`toBeString Custom Matcher array with string value is not a string 1`] = `
"matcherHint mock, args: [".toBeString","received",""]

Expected:
  printExpected mock, args: ["type of string OR instanceof String"]
Received:
  printReceived mock, args: ["object"]"
`;

exports[`toBeString Custom Matcher empty array is not a string 1`] = `
"matcherHint mock, args: [".toBeString","received",""]

Expected:
  printExpected mock, args: ["type of string OR instanceof String"]
Received:
  printReceived mock, args: ["object"]"
`;

exports[`toBeString Custom Matcher extra large number as Bigint object is not a string 1`] = `
"matcherHint mock, args: [".toBeString","received",""]

Expected:
  printExpected mock, args: ["type of string OR instanceof String"]
Received:
  printReceived mock, args: ["bigint"]"
`;

exports[`toBeString Custom Matcher false boolean as object is not a string 1`] = `
"matcherHint mock, args: [".toBeString","received",""]

Expected:
  printExpected mock, args: ["type of string OR instanceof String"]
Received:
  printReceived mock, args: ["boolean"]"
`;

exports[`toBeString Custom Matcher false boolean as primitive is not a string 1`] = `
"matcherHint mock, args: [".toBeString","received",""]

Expected:
  printExpected mock, args: ["type of string OR instanceof String"]
Received:
  printReceived mock, args: ["boolean"]"
`;

exports[`toBeString Custom Matcher false boolean as primitive is not a string 2`] = `
"matcherHint mock, args: [".toBeString","received",""]

Expected:
  printExpected mock, args: ["type of string OR instanceof String"]
Received:
  printReceived mock, args: ["boolean"]"
`;

exports[`toBeString Custom Matcher null is not a string 1`] = `
"matcherHint mock, args: [".toBeString","received",""]

Expected:
  printExpected mock, args: ["type of string OR instanceof String"]
Received:
  printReceived mock, args: ["object"]"
`;

exports[`toBeString Custom Matcher number as Bigint object is not a string 1`] = `
"matcherHint mock, args: [".toBeString","received",""]

Expected:
  printExpected mock, args: ["type of string OR instanceof String"]
Received:
  printReceived mock, args: ["bigint"]"
`;

exports[`toBeString Custom Matcher number as Number object is not a string 1`] = `
"matcherHint mock, args: [".toBeString","received",""]

Expected:
  printExpected mock, args: ["type of string OR instanceof String"]
Received:
  printReceived mock, args: ["object"]"
`;

exports[`toBeString Custom Matcher number as primitive is not a string 1`] = `
"matcherHint mock, args: [".toBeString","received",""]

Expected:
  printExpected mock, args: ["type of string OR instanceof String"]
Received:
  printReceived mock, args: ["number"]"
`;

exports[`toBeString Custom Matcher object with string value is not a string 1`] = `
"matcherHint mock, args: [".toBeString","received",""]

Expected:
  printExpected mock, args: ["type of string OR instanceof String"]
Received:
  printReceived mock, args: ["object"]"
`;

exports[`toBeString Custom Matcher true boolean as object is not a string 1`] = `
"matcherHint mock, args: [".toBeString","received",""]

Expected:
  printExpected mock, args: ["type of string OR instanceof String"]
Received:
  printReceived mock, args: ["boolean"]"
`;

exports[`toBeString Custom Matcher true boolean as primitive is not a string 1`] = `
"matcherHint mock, args: [".toBeString","received",""]

Expected:
  printExpected mock, args: ["type of string OR instanceof String"]
Received:
  printReceived mock, args: ["boolean"]"
`;

exports[`toBeString Custom Matcher undefined is not a string 1`] = `
"matcherHint mock, args: [".toBeString","received",""]

Expected:
  printExpected mock, args: ["type of string OR instanceof String"]
Received:
  printReceived mock, args: ["undefined"]"
`;

exports[`toBeString Custom Matcher zero number as Bigint object is not a string 1`] = `
"matcherHint mock, args: [".toBeString","received",""]

Expected:
  printExpected mock, args: ["type of string OR instanceof String"]
Received:
  printReceived mock, args: ["bigint"]"
`;

exports[`toBeString Custom Matcher zero number as Number object is not a string 1`] = `
"matcherHint mock, args: [".toBeString","received",""]

Expected:
  printExpected mock, args: ["type of string OR instanceof String"]
Received:
  printReceived mock, args: ["object"]"
`;

exports[`toBeString Custom Matcher zero number as primitive is not a string 1`] = `
"matcherHint mock, args: [".toBeString","received",""]

Expected:
  printExpected mock, args: ["type of string OR instanceof String"]
Received:
  printReceived mock, args: ["number"]"
`;
